### PR TITLE
Add playwright to js-environments

### PR DESCRIPTION
**Pull request by gmkumar2005**
Adding details of Playwright based js environment.

### CODE REVIEW

File: doc/project/js-environments.md [hunk 1]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
 For more options of the PhantomJS environment, see
 [the Scaladoc of `PhantomJSEnv`]({{ site.production_url }}/api/sbt-scalajs-env-phantomjs/1.0.0/org/scalajs/jsenv/phantomjs/sbtplugin/PhantomJSEnvPlugin$$autoImport$.html).
 
+## Playwright 
+[Playwright](https://playwright.dev/) is a comprehensive testing library, enabling automation of Chromium, Firefox, and WebKit browsers. It supports multiple platforms and languages, including Mobile Web, making it an optimal choice for testing JavaScript in real browser environments.
```
Comment (sjrd) [inline doc/project/js-environments.md:73]: We keep one sentence per line for easier diffs later on:
```suggestion
[Playwright](https://playwright.dev/) is a comprehensive testing library, enabling automation of Chromium, Firefox, and WebKit browsers.
It supports multiple platforms and languages, including Mobile Web, making it an optimal choice for testing JavaScript in real browser environments. 
```

File: doc/project/js-environments.md [hunk 2]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
 For more options of the PhantomJS environment, see
 [the Scaladoc of `PhantomJSEnv`]({{ site.production_url }}/api/sbt-scalajs-env-phantomjs/1.0.0/org/scalajs/jsenv/phantomjs/sbtplugin/PhantomJSEnvPlugin$$autoImport$.html).
 
+## Playwright 
+[Playwright](https://playwright.dev/) is a comprehensive testing library, enabling automation of Chromium, Firefox, and WebKit browsers. It supports multiple platforms and languages, including Mobile Web, making it an optimal choice for testing JavaScript in real browser environments. 
+
+[`scala-js-env-playwright`](https://github.com/gmkumar2005/scala-js-env-playwright) is an independent project that offers jsEnv and employs Playwright for JavaScript execution.
```
Comment (sjrd) [inline doc/project/js-environments.md:75]:
```suggestion
[`scala-js-env-playwright`](https://github.com/gmkumar2005/scala-js-env-playwright) is an independent project that offers a `JSEnv` that uses Playwright for JavaScript execution.
```

File: doc/project/js-environments.md [hunk 3]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
...
+
+The playwright based `jsEnv` can be enabled by adding following settings in build.sbt 
+```scala
+jsEnv := new PWEnv(
+      browserName = "chrome",
+      headless = true,
+      showLogs = true
+    )
```
Comment (sjrd) [inline doc/project/js-environments.md:83]:
```suggestion
jsEnv := new PWEnv(
  browserName = "chrome",
  headless = true,
  showLogs = true
)
```

File: doc/project/js-environments.md [hunk 4]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
 For more options of the PhantomJS environment, see
 [the Scaladoc of `PhantomJSEnv`]({{ site.production_url }}/api/sbt-scalajs-env-phantomjs/1.0.0/org/scalajs/jsenv/phantomjs/sbtplugin/PhantomJSEnvPlugin$$autoImport$.html).
 
+## Playwright 
+[Playwright](https://playwright.dev/) is a comprehensive testing library, enabling automation of Chromium, Firefox, and WebKit browsers. It supports multiple platforms and languages, including Mobile Web, making it an optimal choice for testing JavaScript in real browser environments. 
+
+[`scala-js-env-playwright`](https://github.com/gmkumar2005/scala-js-env-playwright) is an independent project that offers jsEnv and employs Playwright for JavaScript execution.
+
+The playwright based `jsEnv` can be enabled by adding following settings in build.sbt
```
Comment (sjrd) [inline doc/project/js-environments.md:77]:
```suggestion
The playwright-based `jsEnv` can be enabled by adding the following settings in `build.sbt`
```

File: doc/project/js-environments.md [hunk 5]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
...
+
+The playwright based `jsEnv` can be enabled by adding following settings in build.sbt 
+```scala
+jsEnv := new PWEnv(
+      browserName = "chrome",
+      headless = true,
+      showLogs = true
+    )
+```
+Addtionally it requires following line in `project/plugins.sbt`:
```
Comment (sjrd) [inline doc/project/js-environments.md:85]:
```suggestion
Addtionally it requires the following line in `project/plugins.sbt`:
```

File: doc/project/js-environments.md [hunk 6]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
...
+```scala
+// For Scala.js 1.x
+libraryDependencies += "io.github.gmkumar2005" %% "scala-js-env-playwright" % "0.1.8"
+```
+
+The `browserName` parameter accepts `chrome` , `chromium` , `firefox`, and `webkit` as possible options. Please be aware that webkit support is currently in an experimental stage.
```
Comment (sjrd) [inline doc/project/js-environments.md:91]:
```suggestion
The `browserName` parameter accepts `chrome` , `chromium` , `firefox`, and `webkit` as possible options.
Please be aware that webkit support is currently in an experimental stage.
```

File: doc/project/js-environments.md [hunk 7]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
...
+
+### In browser debugging
+To maintain the browser window open post-execution, simply incorporate the `withKeepAlive`  option into the environment.
```
Comment (sjrd) [inline doc/project/js-environments.md:94]: "simply" never adds real meaning to a sentence; it should be removed:
```suggestion
To maintain the browser window open post-execution, add the `withKeepAlive`  option into the environment.
```

File: doc/project/js-environments.md [hunk 8]
```diff
@@ -69,6 +69,51 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
...
+jsEnv := new jsenv.playwright.PWEnv(
+  browserName = "chrome",
+  headless = true,
+  showLogs = true,
+  pwenvConfig.value,
+)
+
+``` 
+### Headless Usage
+Running in headless mode is crucial for operations within Docker containers and build servers. By default, `scala-js-env-playwright` operates in headless mode. However, for debugging purposes, you can set headless to `false`.
```
Comment (sjrd) [inline doc/project/js-environments.md:112]:
```suggestion
Running in headless mode is crucial for operations within Docker containers and build servers.
By default, `scala-js-env-playwright` operates in headless mode.
However, for debugging purposes, you can set headless to `false`.
```

**Review by sjrd — COMMENTED** [review]
Some remarks mostly on grammar. LGTM otherwise.

**Comment by gmkumar2005** [comment]
All recommendations done

**Review by sjrd — APPROVED** [review]
Thank you!

**Pull request merged**